src/app/core/auth/access-token.resolver.ts

```diff
@@ -0,0 +1,57 @@
+import { inject } from '@angular/core';
+import {
+  ResolveFn,
+  Router,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+import { getForbiddenRoute } from '../../app-routing-paths';
+import { hasValue } from '../../shared/empty.util';
+import { ItemRequestDataService } from '../data/item-request-data.service';
+import { redirectOn4xx } from '../shared/authorized.operators';
+import { ItemRequest } from '../shared/item-request.model';
+import {
+  getFirstCompletedRemoteData,
+  getFirstSucceededRemoteDataPayload,
+} from '../shared/operators';
+import { AuthService } from './auth.service';
+
+/**
+ * Resolve an ItemRequest based on the accessToken in the query params
+ * Used in item-page-routes.ts to resolve the item request for all Item page components
+ * @param route
+ * @param state
+ * @param router
+ * @param authService
+ * @param itemRequestDataService
+ */
+export const accessTokenResolver: ResolveFn<ItemRequest> = (
+  route,
+  state,
+  router: Router = inject(Router),
+  authService: AuthService = inject(AuthService),
+  itemRequestDataService: ItemRequestDataService = inject(ItemRequestDataService),
+): Observable<ItemRequest> => {
+  const accessToken = route.queryParams.accessToken;
+  // Set null object if accesstoken is empty
+  if ( !hasValue(accessToken) ) {
+    return null;
+  }
+  // Get the item request from the server
+  return itemRequestDataService.getSanitizedRequestByAccessToken(accessToken).pipe(
+    getFirstCompletedRemoteData(),
+    // Handle authorization errors, not found errors and forbidden errors as normal
+    redirectOn4xx(router, authService),
```

Comment:
Right now this means that you'll be shown a login page if you try to follow the access link after it expires.

It would be more informative if we could still show the Item page, but with a notice to explain what's going on.